llorens/day03/Day03Solver.py

```diff
@@ -0,0 +1,86 @@
+def calculate_number(matrix_indexes, x, y):
+    res = 0
+    for i in range(-1, 2):
+        for j in range(-1, 2):
+            try:
+                res += matrix_indexes[(x+i, y+j)]
+            except:
+                res += 0
+
+    return  res
+
+
+def solve_part_01(target_num):
+    matrix = {1: (0, 0)}
+    x = 0
+    y = 0
+    num = 1
+    for i in range(1, 1000, 2):
+
+        if num >= target_num:
+            return abs(matrix[target_num][0]) + abs(matrix[target_num][1])
+
+        for desp_x in range(1, i + 1):
+            x += 1
+            num += 1
+            matrix[num] = (x, y)
+
+        for desp_y in range(1, i + 1):
+            y += 1
+            num += 1
+            matrix[num] = (x, y)
+
+        for desp_x in range(1, i + 2):
+            x -= 1
+            num += 1
+            matrix[num] = (x, y)
+
+        for desp_y in range(1, i + 2):
+            y -= 1
+            num += 1
+            matrix[num] = (x, y)
+
```

Comment:
This solution is similar to mine but this time I would say it is cleaner in Python than in Scala 👍

Reply:
Yes very similar, @Cotel Python >>> Scala.

Reply:
I suppose Scala is uglier in this type of algorithms because it is not intended to solve them like this.

There must be another way 

![](https://media.giphy.com/media/SabSYEpsVh0di/giphy.gif)